### ISSUE TITLE
ServerError needs to add some null checking around FillValues

### ISSUE THREAD

**Issue by niemyjski** [elastic/elasticsearch-net#2582]
**Please note I have break on all exceptions on :)**

I'm running latest stable client against 5.1.2 (haven't updated it yet) Code in question: https://github.com/elastic/elasticsearch-net/blob/master/src/Elasticsearch.Net/Responses/ServerError.cs#L157

I'm trying to track down how to reproduce but I started writing a test on a live stream and I got this exception:
```
System.NullReferenceException occurred
 HResult=-2147467261
 Message=Object reference not set to an instance of an object.
 Source=Elasticsearch.Net
 StackTrace:
      at Elasticsearch.Net.RootCauseExtensions.FillValues(IRootCause rootCause, IDictionary`2 dict) in C:\Users\russ\source\elasticsearch-net-5.x\src\Elasticsearch.Net\Responses\ServerError.cs:line 159
 InnerException:
```

I just checked fiddler and it occurs right after a `HEAD` request.
```
HEAD http://127.0.0.1:9200/test-events-v1-%2A HTTP/1.1
Accept: application/json
Content-Type: application/json
Host: 127.0.0.1:9200
```
With the response

```
HTTP/1.1 404 Not Found
content-type: text/plain; charset=UTF-8
content-length: 0
```

This is the callstack when the exception is thrown

```
 	Elasticsearch.Net.dll!Elasticsearch.Net.RootCauseExtensions.FillValues(Elasticsearch.Net.IRootCause rootCause = {Elasticsearch.Net.Error}, System.Collections.Generic.IDictionary<string, object> dict = null) Line 159	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.Error.Create(System.Collections.Generic.IDictionary<string, object> dict, Elasticsearch.Net.IJsonSerializerStrategy strategy = null) Line 94	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.ServerError.Create(System.Collections.Generic.IDictionary<string, object> dict, Elasticsearch.Net.IJsonSerializerStrategy strategy) Line 52	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.SimpleJson.DeserializeObject<Elasticsearch.Net.ServerError>(string json, Elasticsearch.Net.IJsonSerializerStrategy jsonSerializerStrategy) Line 565	C#
>	Elasticsearch.Net.dll!Elasticsearch.Net.ElasticsearchDefaultSerializer.DeserializeAsync<Elasticsearch.Net.ServerError>(System.IO.Stream stream = {System.IO.MemoryStream}, System.Threading.CancellationToken cancellationToken = IsCancellationRequested = false) Line 46	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.ServerError.CreateAsync(System.IO.Stream stream, System.Threading.CancellationToken token) Line 18	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.ServerError.TryCreateAsync(System.IO.Stream stream = {System.IO.MemoryStream}, System.Threading.CancellationToken token = IsCancellationRequested = false)	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.ResponseBuilder<Nest.GetAliasResponse>.SetBodyAsync(Elasticsearch.Net.ElasticsearchResponse<Nest.GetAliasResponse> response = {System.IO.MemoryStream}, System.IO.Stream stream = {Elasticsearch.Net.ElasticsearchResponse<Nest.GetAliasResponse>}) Line 116	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.ResponseBuilder<Nest.GetAliasResponse>.ToResponseAsync() Line 44	C#
 	Elasticsearch.Net.dll!Elasticsearch.Net.HttpConnection.RequestAsync<Nest.GetAliasResponse>(Elasticsearch.Net.RequestData requestData = {Elasticsearch.Net.RequestData}, System.Threading.CancellationToken cancellationToken = IsCancellationRequested = false) Line 205	C#

```

**Comment by niemyjski** [comment]
Looking at the stack trace it's trying to get an alias response but where it's paused under the debugger the last request is the Head request above. Here is the previous request

```
GET http://127.0.0.1:9200/_alias/test-organizations HTTP/1.1
Accept: application/json
Content-Type: application/json
Host: 127.0.0.1:9200
```

```
HTTP/1.1 404 Not Found
content-type: application/json; charset=UTF-8
content-length: 59

{"error":"alias [test-organizations] missing","status":404}
```

**Comment by Mpdreamz** [comment]
Closed by #2586
